internal/cli/cmd/plugin/testschemas/test-schemas.go

```diff
@@ -0,0 +1,531 @@
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		if filepath.Ext(currentPath) != ".json" {
+			return nil
+		}
+
+		testName := filepath.Base(currentPath)
+		// Calculate relative path from schemas directory
+		relPath, err := filepath.Rel(o.cfg.SchemasPath, currentPath)
+		if err != nil {
+			relPath = currentPath // Fallback to full path if relative path calculation fails
+		}
+		result := TestResult{
+			TestName: testName,
+			TestPath: relPath,
+			TestType: map[bool]TestType{true: TestTypeModelValid, false: TestTypeModelInvalid}[shouldBeValid],
+		}
+
+		// Read test data
+		data, err := os.ReadFile(currentPath) //nolint: gosec
+		if err != nil {
+			result.Error = fmt.Sprintf("Failed to read test file: %v", err)
+			results = append(results, result)
+			return nil
+		}
+
+		// Unmarshal the JSON to the plugin struct
+		var plugin common.Plugin
+		if err := json.Unmarshal(data, &plugin); err != nil {
+			result.Error = fmt.Sprintf("Failed to parse JSON: %v", err)
+			results = append(results, result)
+			return nil
+		}
+
+		// Validate the plugin against the schema
+		ctx := cuecontext.New()
+		pluginValue := ctx.CompileBytes(data)
+		finalValue := pluginValue.Unify(ctx.BuildInstance(buildInstance))
+
+		logrus.Debugf("Running model validation for %s", currentPath)
+		validationErr := finalValue.Validate(schema.CueValidationOptions...)
+		isValid := validationErr == nil
+
+		if !isValid {
+			logrus.Debugf("Evaluation error for %s: %v", testName, validationErr)
+		}
+
+		if shouldBeValid {
+			result.Success = isValid
+			if !isValid {
+				result.Error = fmt.Sprintf("Expected valid data but validation failed: %v", validationErr)
+			}
+		} else {
+			result.Success = !isValid
+			if isValid {
+				result.Error = "Expected invalid data but validation succeeded"
+			}
+		}
+
+		results = append(results, result)
+		return nil
+	})
+}
+
+// runMigrationTestsForPath runs migration tests for a specific directory path
+func (o *option) runMigrationTestsForPath(testDir string, buildInstance *build.Instance, pluginKind v1plugin.Kind) ([]TestResult, error) {
+	var results []TestResult
+
+	return results, filepath.WalkDir(testDir, func(currentPath string, d os.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if !d.IsDir() {
+			return nil
+		}
+
+		// Skip the root migrate directory
+		if currentPath == testDir {
+			return nil
+		}
+
+		testName := d.Name()
+
+		relPath, err := filepath.Rel(o.cfg.SchemasPath, currentPath)
+		if err != nil {
+			relPath = currentPath // Fallback to full path if relative path calculation fails
+		}
+
+		result := TestResult{
+			TestName: testName,
+			TestPath: relPath,
+			TestType: TestTypeMigrate,
+		}
+
+		// Look for input.json and expected.json in this directory
+		inputPath := filepath.Join(currentPath, "input.json")
+		expectedPath := filepath.Join(currentPath, "expected.json")
+
+		if _, err := os.Stat(inputPath); os.IsNotExist(err) {
```

Comment:
`os.IsNotExist` will only check if the error is matching the "not existing file". But if there is any other error, this check will be false but you will still have an issue with the file.

An alternative would be to do something like that:

```go
if _, err := os.Stat(inputPath); err != nil {
	result.Error = fmt.Sprintf("issue while opening %q: %s", inputPath, err),
	results = append(results, result)
	return filepath.SkipDir
}
```

Reply:
actually I would just remove these pre-checks. Reading the file will fail immediately if the file does not exist with an accurate error. So it's enough